src/shared/socket-netlink.c

```diff
@@ -235,6 +235,76 @@ int socket_address_parse_and_warn(SocketAddress *a, const char *s) {
         return 0;
 }
 
+int socket_addr_port_from_string_auto(const char *s,  uint16_t port, SocketAddress *a) {
```

Comment:
We have `in_addr_port_ifindex_name_from_string_auto()`. No need to reimplement the parser again. Just make it use the shared implementation and pass NULL for the parts which cannot be specified.

Reply:
yes I reused that as a wrapper for socket. thanks